### PR TITLE
Improved logging for cli tools

### PR DESCRIPTION
**Pull request by Vovchyk**
## Description

This improves logging for cli tools (a.k.a developer tools), which should be distinguished from the `RskJ` node and mostly are intended for development purposes.

This also adds a capability to specify a log level in command line (via JVM args) for the `logs/rsk.log` file (the primary logs output), stdout (console output) and separately for stdout of the cli tools. For instance, in the command bellow the log level for the file is `DEBUG`(`-Dlogging=DEBUG`), for `stdout` - `ERROR`(`-Dlogging.stdout=ERROR`), and for `stdout.clitools` - `INFO`(`-Dlogging.stdout.clitools=INFO`):
```bash
java -Dlogging=DEBUG -Dlogging.stdout=ERROR -Dlogging.stdout.clitools=INFO -cp ./rskj-core-3.1.0-SNAPSHOT-all.jar co.rsk.cli.tools.RewindBlocks 1000000 --testnet
```

If not specified, the following are default log levels:
```
-Dlogging=DEBUG
-Dlogging.stdout=OFF
-Dlogging.stdout.clitools=INFO
```

This capability allows to specify log level not only for a cli tool, but also for the rskj node itself. E.g. the following command sets log level `INFO` for the file and `WARN` - for the `stdout`:
```
java -Dlogging=INFO -Dlogging.stdout=WARN -jar ./rskj-core-3.1.0-SNAPSHOT-all.jar --testnet
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Comment by Vovchyk** [comment]
pipeline:run

**Review by rmoreliovlabs — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/cli/CliToolRskContextAware.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    }
+
+    public void execute(@Nonnull String[] args) {
+        execute(args, () -> new RskContext(args), System::exit);
+    }
+
+    public void execute(@Nonnull String[] args, @Nonnull Factory<RskContext> contextFactory, @Nonnull NodeStopper stopper) {
+        Objects.requireNonNull(args, "args should not be null");
+        Objects.requireNonNull(contextFactory, "contextFactory should not be null");
```
Comment (joaquinlpereyra) [inline rskj-core/src/main/java/co/rsk/cli/CliToolRskContextAware.java:60]: If `stopper` is null then `NPE` on line 69 _after_ execution, seems reasonable to be consistent and require-non-null for that too.

File: rskj-core/src/main/java/co/rsk/cli/CliToolRskContextAware.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * An abstract class for cli tools that need {@link RskContext}. Lifecycle of the  {@link RskContext} instance
+ * is being managed by {@link CliToolRskContextAware}.
+ *
+ * Also {@link CliToolRskContextAware} provides a logger instance for derived classes.
+ */
+public abstract class CliToolRskContextAware {
```
Comment (joaquinlpereyra) [inline rskj-core/src/main/java/co/rsk/cli/CliToolRskContextAware.java:35]: More of a question than a comment: why choose this as a sort of wrapper class? Wouldn't it be easier for CLI tools to just inherit from this class? `execute` would work mostly the same, calling an abstract method that must be implemented.

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Review by heitorfm — APPROVED** [review]

**Pull request merged**